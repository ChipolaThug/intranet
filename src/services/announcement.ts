export class Announcement {
    id = 0

    created = new Date()

    type = ""

    title = ""

    url = ""
}

export const translationAnnouncement: { [k in keyof Announcement]: string } = {
    id: "id",
    created: "creation",
    type: "type",
    title: "titre",
    url: "url",
}

export const elementName = "Announcement"

export type AnnouncementWithoutId = Omit<Announcement, "id">
