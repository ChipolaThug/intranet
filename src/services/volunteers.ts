export class Volunteer {
    id = 0

    lastname = ""

    firstname = ""

    email = ""

    mobile = ""

    photo = ""

    adult = 1

    roles: string[] = []

    active = ""

    discordId = ""

    dayWishes: string[] = []

    dayWishesComment = ""

    age = 0

    teeshirtSize = ""

    food = ""

    teamWishes: number[] = []

    teamWishesComment = ""

    hiddenNotifs: number[] = []

    created = new Date()

    password1 = ""

    password2 = ""

    pushNotifSubscription = ""

    acceptsNotifs = ""
}

export const translationVolunteer: { [k in keyof Volunteer]: string } = {
    id: "id",
    lastname: "nom",
    firstname: "prenom",
    email: "mail",
    mobile: "telephone",
    photo: "photo",
    adult: "majeur",
    roles: "roles",
    active: "actif",
    discordId: "discordId",
    dayWishes: "enviesJours",
    dayWishesComment: "commentaireEnviesJours",
    age: "age",
    teeshirtSize: "teeshirt",
    food: "alimentation",
    teamWishes: "enviesEquipe",
    teamWishesComment: "commentaireEnviesEquipe",
    hiddenNotifs: "notifsCachees",
    created: "creation",
    password1: "passe1",
    password2: "passe2",
    pushNotifSubscription: "pushNotifSubscription",
    acceptsNotifs: "accepteLesNotifs",
}

export const elementName = "Volunteer"

export const volunteerExample: Volunteer = {
    id: 1,
    firstname: "Aupeix",
    lastname: "Amélie",
    email: "pakouille.lakouille@yahoo.fr",
    mobile: "0675650392",
    photo: "images/volunteers/$taille/amélie_aupeix.jpg",
    adult: 1,
    roles: [],
    active: "inconnu",
    discordId: "",
    dayWishes: [],
    dayWishesComment: "",
    age: 33,
    teeshirtSize: "FM",
    food: "Végétarien",
    teamWishes: [],
    teamWishesComment: "",
    hiddenNotifs: [],
    created: new Date(0),
    password1: "$2y$10$fSxY9AIuxSiEjwF.J3eXGubIxUPkdq9d5fqpbl8ASimSjNj4SR.9O",
    password2: "$2y$10$fSxY9AIuxSiEjwF.J3eXGubIxUPkdq9d5fqpbl8ASimSjNj4SR.9O",
    pushNotifSubscription: "",
    acceptsNotifs: "",
}

export const emailRegexp =
    /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i
export const passwordMinLength = 4

export type VolunteerWithoutId = Omit<Volunteer, "id">

export interface VolunteerLogin {
    jwt: string
    id: number
    roles: string[]
}

export interface VolunteerForgot {
    message: string
}

export interface VolunteerNotifs {
    id: Volunteer["id"]
    firstname: Volunteer["firstname"]
    adult: Volunteer["adult"]
    active: Volunteer["active"]
    hiddenNotifs: Volunteer["hiddenNotifs"]
    pushNotifSubscription: Volunteer["pushNotifSubscription"]
    acceptsNotifs: Volunteer["acceptsNotifs"]
}

export interface VolunteerTeamWishes {
    id: Volunteer["id"]
    teamWishes: Volunteer["teamWishes"]
    teamWishesComment: Volunteer["teamWishesComment"]
}

export interface VolunteerDayWishes {
    id: Volunteer["id"]
    dayWishes: Volunteer["dayWishes"]
    dayWishesComment: Volunteer["dayWishesComment"]
}

export interface VolunteerParticipationDetails {
    id: Volunteer["id"]
    age: Volunteer["age"]
    teeshirtSize: Volunteer["teeshirtSize"]
    food: Volunteer["food"]
}
