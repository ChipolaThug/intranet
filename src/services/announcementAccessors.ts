import ServiceAccessors from "./accessors"
import { elementName, Announcement, AnnouncementWithoutId } from "./announcement"

const serviceAccessors = new ServiceAccessors<AnnouncementWithoutId, Announcement>(elementName)

// export const announcementGet = serviceAccessors.get()
// export const announcementAdd = serviceAccessors.add()
// export const announcementSet = serviceAccessors.set()

export const announcementListGet = serviceAccessors.securedListGet()
