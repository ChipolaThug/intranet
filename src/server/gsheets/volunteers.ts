import getExpressAccessors from "./expressAccessors"
import { sheetNames } from "./accessors"
import { Volunteer, VolunteerWithoutId, translationMember } from "../../services/volunteers"

const { listGetRequest, getRequest, setRequest, addRequest } = getExpressAccessors<
    VolunteerWithoutId,
    Volunteer
>(sheetNames.Volunteers, new Volunteer(), translationMember)

export const volunteerListGet = listGetRequest()

export const volunteerGet = getRequest()

export const volunteerAdd = addRequest()

export const volunteerSet = setRequest()
