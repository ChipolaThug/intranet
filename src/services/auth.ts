import { AxiosRequestConfig } from "axios"
import Cookies from "js-cookie"

import { VolunteerLogin } from "./volunteers"

export const axiosConfig: AxiosRequestConfig = {
    headers: {},
}

export function setJWT(token: string, id: number): void {
    axiosConfig.headers.Authorization = `Bearer ${token}`
    Cookies.set("jwt", token, { expires: 3650 })
    Cookies.set("id", `${id}`, { expires: 3650 })
}

export function unsetJWT(): void {
    delete axiosConfig.headers.Authorization

    Cookies.remove("jwt")
    Cookies.remove("id")
}

export function getCookieJWT(cookie = ""): VolunteerLogin {
    const cookies = cookie
        .split(";")
        .reduce((res: { [cookieName: string]: string }, el: string) => {
            const [k, v] = el.split("=")
            res[k.trim()] = v
            return res
        }, {})
    return { jwt: cookies.jwt, id: +cookies.id }
}
