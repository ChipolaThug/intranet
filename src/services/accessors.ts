import axios from "axios"

import config from "../config"
import { axiosConfig } from "./auth"

export type ElementWithId = unknown & { id: number }

export type ElementTranslation<Element> = { [k in keyof Element]: string }

export default function getServiceAccessors<
    // eslint-disable-next-line @typescript-eslint/ban-types
    ElementNoId extends object,
    Element extends ElementNoId & ElementWithId
>(elementName: string): any {
    function get(): (id: number) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (id: number): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.get(`${config.API_URL}/${elementName}Get`, {
                    ...axiosConfig,
                    params: { id },
                })
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function listGet(): () => Promise<{
        data?: Element[]
        error?: Error
    }> {
        interface ElementListGetResponse {
            data?: Element[]
            error?: Error
        }
        return async (): Promise<ElementListGetResponse> => {
            try {
                const { data } = await axios.get(
                    `${config.API_URL}/${elementName}ListGet`,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    // eslint-disable-next-line @typescript-eslint/ban-types
    function add(): (volunteerWithoutId: ElementNoId) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (volunteerWithoutId: ElementNoId): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.post(
                    `${config.API_URL}/${elementName}Add`,
                    volunteerWithoutId,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function set(): (volunteer: Element) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (volunteer: Element): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.post(
                    `${config.API_URL}/${elementName}Set`,
                    volunteer,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function countGet(): () => Promise<{
        data?: number
        error?: Error
    }> {
        interface ElementCountGetResponse {
            data?: number
            error?: Error
        }
        return async (): Promise<ElementCountGetResponse> => {
            try {
                const { data } = await axios.get(
                    `${config.API_URL}/${elementName}CountGet`,
                    axiosConfig
                )
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    function customPost(apiName: string): (params: any) => Promise<{
        data?: Element
        error?: Error
    }> {
        interface ElementGetResponse {
            data?: Element
            error?: Error
        }
        return async (params: any): Promise<ElementGetResponse> => {
            try {
                const { data } = await axios.post(
                    `${config.API_URL}/${elementName}${apiName}`,
                    params,
                    axiosConfig
                )
                if (data.error) {
                    throw Error(data.error)
                }
                return { data }
            } catch (error) {
                return { error: error as Error }
            }
        }
    }

    return { listGet, get, set, add, countGet, customPost }
}
