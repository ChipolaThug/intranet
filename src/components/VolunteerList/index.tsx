import { memo } from "react"
import { Link } from "react-router-dom"

import { Volunteer } from "../../services/volunteers"
import styles from "./styles.module.scss"

interface Props {
    items: Volunteer[]
}

const VolunteerList = ({ items }: Props) => (
    <div className={styles["user-list"]}>
        <h4>Volunteer List</h4>
        <ul>
            {items.map(({ id, lastname, firstname }) => (
                <li key={id}>
                    <Link to={`/Volunteer/${id}`}>
                        <b>{firstname}</b> {lastname}
                    </Link>
                </li>
            ))}
        </ul>
    </div>
)

export default memo(VolunteerList)
