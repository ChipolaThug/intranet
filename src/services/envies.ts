import { get, listGet, add, set } from "./accessors"

export class Envie {
    id = 0

    domaine = ""

    envies = ""

    precisions = ""

    equipes: string[] = []

    dateAjout = ""
}

export type EnvieWithoutId = Omit<Envie, "id">

export const envieGet = get<Envie>("Envie")

export const envieListGet = listGet<Envie>("Envie")

export const envieAdd = add<EnvieWithoutId, Envie>("Envie")

export const envieSet = set<Envie>("Envie")
