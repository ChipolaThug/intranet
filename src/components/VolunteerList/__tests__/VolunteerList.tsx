/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

import { volunteerExample } from "../../../services/volunteers"
import VolunteerList from "../index"

describe("<VolunteerList />", () => {
    it("renders", () => {
        const tree = render(
            <MemoryRouter>
                <VolunteerList items={[volunteerExample]} />
            </MemoryRouter>
        ).container.firstChild

        expect(tree).toMatchSnapshot()
    })
})
