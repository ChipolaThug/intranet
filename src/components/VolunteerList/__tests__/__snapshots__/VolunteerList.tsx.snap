// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VolunteerList /> renders 1`] = `
<div
  class="VolunteerList"
>
  <h4>
    Volunteer List
  </h4>
  <ul>
    <li>
      <a
        href="/Volunteer/1"
      >
        <b>
          Aupeix
        </b>
         
        Am√©lie
      </a>
    </li>
  </ul>
</div>
`;
